File: backend/src/bds_backend/schemas/teams.py
"""STUB — not implemented. Copy the pattern from schemas/auth.py.

You'll want at minimum:

    class TeamCreate(BaseModel):
        name: str = Field(min_length=1, max_length=150)
        sport_id: uuid.UUID
        season: str

    class TeamRead(BaseModel):
        id: uuid.UUID
        name: str
        sport: str          # the sport's NAME, not just its id — joined
                             # in at the DAO level so the frontend doesn't
                             # need a second request just to show it
        season: str
        model_config = {"from_attributes": True}

Note `TeamRead.sport` above: a response schema is free to shape data
differently from the underlying table — it exists to serve the API
consumer, not to mirror the database 1:1. See schemas/auth.py's WHY
note on request vs response schemas for the same idea applied to hiding
`password_hash`.
"""

File: frontend/src/api/teams.ts
/**
 * STUB — not implemented. Copy the pattern from api/auth.ts.
 *
 *   export interface Sport {
 *     id: string;
 *     name: string;
 *   }
 *
 *   export interface Team {
 *     id: string;
 *     name: string;
 *     sport: string;       // matches backend schemas/teams.py's TeamRead —
 *                           // the sport's NAME, already joined server-side
 *     season: string;
 *   }
 *
 *   export async function getSports(): Promise<Sport[]> {
 *     const { data } = await apiClient.get<Sport[]>('/sports');
 *     return data;
 *   }
 *
 *   export async function getTeams(sportId?: string): Promise<Team[]> {
 *     const { data } = await apiClient.get<Team[]>('/teams', {
 *       params: { sport_id: sportId },
 *     });
 *     return data;
 *   }
 *
 * Every field name here must match the backend's Pydantic schema
 * exactly (see backend/.../schemas/teams.py's stub) — this file IS the
 * frontend's copy of that contract, kept in sync by hand.
 */

export {};
